Steps4Impact/NotificationsV2/NotificationsViewController.swift

```diff
@@ -31,35 +31,96 @@ import UIKit
 import RxSwift
 
 class NotificationsViewController: TableViewController {
+  
   override func commonInit() {
     super.commonInit()
 
     title = Strings.Notifications.title
     dataSource = NotificationsDataSource()
   }
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    checkNotificationPermission()
```

Comment:
Because this check decides whether to show the "Stay in Loop permission prompt" or not. So I think its better to make the check before and then hide/show the prompt if needed